### PR TITLE
fixes #749

### LINKED ISSUE #749: Fix Titleize
There is currently a method called `titleize` that lives in `/lib/titleize.js`.
It is incomplete and needs to be fixed!

Ex.
man from the boondocks -> Man From The Boondocks

### Acceptance Criteria:
1. Running `yarn test titleize` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by jsjain007**
Closes #theRelatedIssueNumber

- [X] Running `yarn lint` does not trigger any linter errors
- [X] The test of the method you have fixed is passing
- [X] You have written a new skeleton method for someone else to work on!
- [X] You have written tests to accompany your skeleton method

### CODE REVIEW

File: lib/titleize.js [hunk 1]
```diff
@@ -6,7 +6,12 @@
 */
 
 function titleize(str) {
-  return "Not yet implemented";
+  if(typeof str !== "string") {
+    throw new TypeError();
+  }
+  let regExp = /_|-/gi;
+  str = str.replace(regExp, " ");
```
[the remark this answers is not in the document]
Comment (jsjain007) [reply lib/titleize.js]: Oh, Thanks for pointing it out to me. I will fix it. So should I first make a copy of it them modify it ?

**Comment by jsjain007** [comment]
Hi @ktilcu  @BlakeGuilloud  are there any more changes that I have to make for it to be merged ?

**Comment by ktilcu** [comment]
@jsjain007 Looks good Thanks!
